### ISSUE TITLE
format on display - format on edit

### ISSUE THREAD

**Issue by dpatte** [JackAdams/meteor-editable-text#16]
Most of the fields that I want to make editable have existing helpers that do minimal transformations on the data when displayed on the page. For example, in my application, several of the numeric fields are displayed using helpers such as the following

numHelper: function() { return this.num.toLocaleString(); } // which adds commas

I would like to format my number when being displayed using the above helper, but display the number without formatting while editing. Is that possible?

**Comment by JackAdams** [comment]
At the moment it's not possible.  You can set the displayed value as a formatted string using `value=numHelper` and the function above, but when you edit it will also be formatted.

**Comment by dpatte** [comment]
Most of what i am dealing with here are numbers. They are stored in mongo as numbers, but displayed after editing with commas (1,000). Using before and after hooks, I suspect I should be able to handle this.

**Comment by JackAdams** [comment]
Okay, just released a new version with an undocumented parameter `editingValue`.  Now you can use a helper to manipulate the `value` and `editingValue` separately but, like you say, you'll probably need a `beforeUpdate` hook to transform the editing value into the sort of value you want to store in the db.

**Comment by dpatte** [comment]
Thank you for you help on this, but I don't seem to understand how to use it.
my numeric field in the doc is called cap
when i display it I use a helper (capH) which adds commas to the number while displaying it
but when I click to edit it I want to display cap as it comes from the db (without commas)

I tried the following:

{{> editableText collection="deals" field="cap" value=capH editingValue=cap acceptEmpty=true}}

I expected it to show cap as the value when editing, but clicking on the field to edit simply displays the number (but now editable) with commas as it did before I edited it. It seems to ignore whatever I have defined for editingValue.
